walkthroughs/1C-managing-and-protecting-using-api-mgmt/walkthrough.adoc

```diff
@@ -0,0 +1,381 @@
+
+[type=verification]
+Click *Visit Portal* under the *Developer Portal* section on the side navigation and click *Documentation* in the top navigation. Can you see the visual representation of the Orders API specification?
+
+[type=verificationFail]
+Verify that you followed each step in the procedure above. If you are still having issues, contact your administrator.
+
+[time=10]
+== Make API calls through the Developer Portal
+
+. Log in to the Developer Portal:
+.. Go to the link:https://{user-sanitized-username}.{openshift-app-host}[3scale Developer Portal, window="_blank"].
+.. Close the side bar on the right, if it is open.
+.. Click *Sign in* in the top navigation bar.
+.. Enter `john` as *Username or Email* and `123456` as *Password*. These are the credentials of the _Developer_ account created by default for each tenant.
+.. Click *Sign in*.
++
+. Get the application credentials:
+.. Click *API Credentials* in the top navigation bar.
+.. Select *< Applications*.
+.. Click on the `orders-api-app` name.
+.. You will see the application credentials – Client ID and Client Secret.
+.. In the *Redirect URL* field enter `https://{user-sanitized-username}.{openshift-app-host}/oauth2-redirect.html`. This will be the Redirect URL for the OAuth handshake, and it points to the HTML page on the Developer Portal we created in a previous step.
+.. Click *Submit* to save the new redirect URL. Zync will update the Redirect URL of the corresponding client in Red Hat Single Sign-On.
+.. Take note of the Client ID and Client Secret.
++
+. Make API calls through the documentation page without credentials:
+.. Click *Documentation* in the top navigation bar. You should see the visualized Orders API documentation.
+.. Expand the `GET /orders/history` and click *Try it out* and then *Execute*.
+.. You should get an error with code `403` and response body `Authentication parameters missing`. This is normal, because the API gateway is expecting the `Authorization` header with the access token, but it is not provided.
++
+. Authenticate with Red Hat Single Sign-On:
+.. Click *Authorize* button.
+.. In the modal window that opens, enter the Client ID and the Client Secret of the application `orders-api-app` in the corresponding fields `CLIENT_ID` and `CLIENT_SECRET`.
+.. Select the `OPENID` checkbox.
+.. Click *Authorize*.
+.. If a Red Hat Single Sign-On login screen is shown with `{user-sanitized-username} Realm` title, enter `{user-sanitized-username}` and `password` as username and password accordingly. It might not be shown, if you already logged in and the session has not expired.
+.. Click *Log In*. You will be redirected back ot the authorization modal window on the developer portal.
+.. Click *Close*.
+.. Under `GET /orders/history`, click *Execute* again (expand it and click *Try it out* if you don't see the *Execute* button).
+.. You should now receive a successful response with status code `200` and the list of orders in the response body.
```

Comment:
The one downside here is that the orders returned here are based on a session ID gotten via cookie, so you need to:

1. create an order
1. note the cookie in the response
1. use that cookie in the /orders/history call

I think this current example will always return an empty JSON array? Not an issue with your instruction, more so with the fact sessions are used.

Reply:
Oooh, OK, I didn't realize that...
It actually returns a `200 OK` with no body, but when I tried that I thought it was because I didn't create any orders after the environment restart.

I think that ideally, the Orders API should keep the orders linked to the username (even if it's still a memory store). The username then can be extracted from the session (if the Cookie is present in the request), or from the JWT token (if no cookie, but `Authorize: Bearer` is present).

WDYT?

Reply:
Yeah, that's a good solution 👍Won't get it done before the lab today, but we can make the change for future labs.